src/project/projectSlice.ts

```diff
@@ -15,42 +15,100 @@
  * along with this program. If not, see https://www.gnu.org/licenses/.
  */
 
-import { createSlice } from '@reduxjs/toolkit';
+import { createAction, createSlice } from '@reduxjs/toolkit';
+import { setUsers, User } from 'terraso-client-shared/account/accountSlice';
+import {
+  Membership,
+  setMembers,
+} from 'terraso-client-shared/memberships/membershipsSlice';
 import * as projectService from 'terraso-client-shared/project/projectService';
-import { createAsyncThunk } from 'terraso-client-shared/store/utils';
+import { setSites, Site } from 'terraso-client-shared/site/siteSlice';
+import {
+  createAsyncThunk,
+  dehydrated,
+} from 'terraso-client-shared/store/utils';
+
+const { plural: dehydrateProjects, sing: dehydrateProject } = dehydrated<
+  Project,
+  HydratedProject
+>({
+  memberships: setMembers,
+  users: setUsers,
+  sites: setSites,
+});
+
+export type SerializableSet = Record<string, boolean>;
 
 export type Project = {
   id: string;
   name: string;
   privacy: 'PRIVATE' | 'PUBLIC';
   description: string;
   updatedAt: string; // this should be Date.toLocaleDateString; redux can't serialize Dates
-  userCount: number;
-  siteCount: number;
+  membershipIds: Record<string, { user: string }>; // TODO: Why doesn't the membership have user info? have to store user id here as well
+  siteIds: SerializableSet;
+  groupId: string;
+};
+
+export type HydratedProject = {
+  dehydrated: Project;
```

Comment:
Suggestion: Do you have to name it `dehydrated` it seems that calling it project will make more sense, or why do we explicitly need to name it `dehydrated` here?